include/fmt/format.h

```diff
@@ -3395,6 +3395,43 @@ inline typename buffer_context<Char>::type::iterator format_to(
                               basic_format_args<context>(as));
 }
 
+namespace internal {
+
+template<typename... Ts>
+struct void_ { typedef void type; };
+
+template <typename T, typename Enable = void_<>::type>
+struct it_category { typedef int type; };
+
+template <typename T>
+struct it_category<T*> { typedef std::output_iterator_tag type; };
+
+template <typename T>
+struct it_category<const T*> { typedef int type; };
+
+template <typename T>
+struct it_category<T, typename void_<typename T::iterator_category>::type> {
+  typedef typename T::iterator_category type;
+};
+
+template <typename It>
+class is_output_iterator {
+  //check for mutability
+  template <typename U>
+  static decltype(*(internal::declval<U>())) test(std::input_iterator_tag);
```

Comment:
Why not just return `const char&`?

Reply:
Makes sense. I was confused by the use of `input_iterator_tag` in the write context, but since iterator tags form a class hierarchy it makes sense now.